### PR TITLE
[Video] Captions and alt text

### PR DESCRIPTION
**Pull request by mozzius**
# Stacked on #4881 and #4884 

Alt text and caption dialog. Captions are only available on web. Max 4 languages.

<table>
  <tr>
    <td>Web composer</td>
    <td>Web dialog open</td>
    <td>Web with captions</td>
  </tr>
  <tr>
    <td>Screenshot 2024-08-28 at 17 05 00</td>
    <td>Screenshot 2024-08-28 at 17 05 48</td>
    <td>Screenshot 2024-08-28 at 17 10 41</td>
  </tr>
</table>

<table>
  <tr>
    <td>iOS composer</td>
    <td>iOS dialog open</td>
  </tr>
  <tr>
    <td>Simulator Screenshot - iPhone 15 - 2024-08-28 at 17 11 53</td>
    <td>Simulator Screenshot - iPhone 15 - 2024-08-28 at 17 11 58</td>
  </tr>
</table>

### CODE REVIEW

**Comment by mozzius** [comment]
Oh, you might see languages with `<Name> ( )` - this is related to the language cleanup PR from earlier

File: src/view/com/composer/videos/SubtitleDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+}
+
+export function SubtitleDialogBtn(props: Props) {
+  const control = Dialog.useDialogControl()
+  const {_} = useLingui()
+
+  return (
+    <View style={[a.flex_row, a.mt_xs]}>
+      <Button
+        label={_(msg`Open video captions dialog`)}
```
Comment (haileyok) [inline src/view/com/composer/videos/SubtitleDialog.tsx:39]: Should this have a different label if only alt is available?

File: src/view/com/composer/videos/SubtitleDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+          file,
+        },
+      ])
+    },
+    [setCaptions, primaryLanguage],
+  )
+
+  const subtitleMissingLanguage = captions.some(sub => sub.lang === '')
+
+  console.log({subtitleMissingLanguage})
```
Comment (haileyok) [inline src/view/com/composer/videos/SubtitleDialog.tsx:86]: log

**Review by haileyok — APPROVED** [review]
Code all looks good. Didn't test with a VTT file but the dialog seems to work, properly shows available options on web/native, etc. Great!!

**Pull request merged**